modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java

```diff
@@ -168,15 +168,23 @@ public long getObjectDefinitionId() {
 
 	public JSONObject getObjectFieldJSONObject(ObjectField objectField) {
 		return JSONUtil.put(
+			// TODO: return indexedLanguageId as null instead of ""
+			"indexedLanguageId", objectField.getIndexedLanguageId()
```

Comment:
@carolmariaabb can you help me with that?

Reply:
Hey, @miltonmc. Using this `JSONUtil`, we can't create a field with `null` value. If we put `null`, the key "indexedLanguageId" will be removed from the final json. Does that work for you?

Reply:
@miltonmc Why do you need to return null instead of empty string?

Reply:
The idea was keep the consistency...

Since the frontend is setting  indexedLanguageId to null before save (in the case of no language selected), I was expecting retrieve the same value into the  editing side panel.

Reply:
This behaviour leaded me to a bug (already solved) and can lead others to the same mistake

Reply:
It's returning "" because of this auto-generated code [here](https://github.com/carolmariaabb/liferay-portal/blob/4bc7f074155fae13178dc677ef75b5c25b60fe4d/modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectFieldModelImpl.java#L766-L772). I don't think we can change this.

We could do something like:
```
String indexedLanguageId = objectField.getIndexedLanguageId();

if (indexedLanguageId.isEmpty()) {
     indexedLanguageId = null;
}
```
before calling `JSONUtil` (but we'll have the behavior that Paulino mentioned above) or we can save an empty string instead of null in case of no language selected.